internal/controller/capicluster_controller.go

```diff
@@ -56,11 +57,9 @@ func (r *CapiClusterReconciler) Reconcile(ctx context.Context, req ctrl.Request)
 	cluster := &capi.Cluster{}
 	err = r.Client.Get(ctx, types.NamespacedName{Name: req.Name, Namespace: req.Namespace}, cluster)
 	if err != nil {
-		// TODO(theo): might need to ignore when objects are not found since we cannot do anything
-		//             see https://book.kubebuilder.io/reference/using-finalizers.html
-		//if r.Client.IsNotFound(err) {
-		//	return ctrl.Result{}, nil
-		//}
+		if apimachineryerrors.IsNotFound(err) {
```

Comment:
We probably want to avoid unecessary if here and use https://github.com/kubernetes-sigs/controller-runtime/blob/5c217307ad12f39d69011306f3f69ec0aebbe48a/pkg/client/interfaces.go#L208 instead so you`d replace 

```go
if apimachineryerrors.IsNotFound(err) {
  return ctrl.Result{}, nil
}
return ctrl.Result{}, errors.WithStack(err)
```

with 

```go
return ctrl.Result{}, client.IgnoreNotFound(errors.WithStack(client.IgnoreNotFound(err))
```

Reply:
But that's also harder to read